simtools/simtel/simtel_runner_ray_tracing.py

```diff
@@ -0,0 +1,256 @@
+
+        # File location
+        self._baseDirectory = io.getOutputDirectory(
+            self._filesLocation,
+            self.label,
+            'ray-tracing'
+        )
+        self._baseDirectory.mkdir(parents=True, exist_ok=True)
+
+        self._singleMirrorMode = singleMirrorMode
+
+        # RayTracing - default parameters
+        self._repNumber = 0
+        self.RUNS_PER_SET = 1 if self._singleMirrorMode else 20
+        self.PHOTONS_PER_RUN = 10000
+
+        # Loading configData
+        _configDataIn = gen.collectDataFromYamlOrDict(configFile, configData)
+        _parameterFile = io.getDataFile('parameters', 'simtel-runner-ray-tracing_parameters.yml')
+        _parameters = gen.collectDataFromYamlOrDict(_parameterFile, None)
+        self.config = gen.validateConfigData(_configDataIn, _parameters)
+
+        self._loadRequiredFiles()
+
+    def _loadRequiredFiles(self):
+        '''
+        Which file are required for running depends on the mode.
+        Here we define and write some information into these files. Log files are always requires.
+        '''
+
+        self._corsikaFile = self._simtelSourcePath.joinpath('run9991.corsika.gz')
+
+        # Loop to define and remove existing files.
+        # Files will be named _baseFile = self.__dict__['_' + base + 'File']
+        for baseName in ['stars', 'photons', 'log']:
+            fileName = names.rayTracingFileName(
+                self.telescopeModel.site,
+                self.telescopeModel.name,
+                self.config.sourceDistance,
+                self.config.zenithAngle,
+                self.config.offAxisAngle,
+                self.config.mirrorNumber if self._singleMirrorMode else None,
+                self.label,
+                baseName
+            )
+            file = self._baseDirectory.joinpath(fileName)
+            if file.exists():
+                file.unlink()
+            # Defining the file name variable as an class atribute.
+            self.__dict__['_' + baseName + 'File'] = file
+
+        # Adding header to photon list file.
+        with self._photonsFile.open('w') as file:
+            file.write('#{}\n'.format(50 * '='))
+            file.write('# List of photons for RayTracing simulations\n')
+            file.write('#{}\n'.format(50 * '='))
+            file.write('# configFile = {}\n'.format(self.telescopeModel.getConfigFile()))
+            file.write('# zenithAngle [deg] = {}\n'.format(self.config.zenithAngle))
+            file.write('# offAxisAngle [deg] = {}\n'.format(self.config.offAxisAngle))
+            file.write('# sourceDistance [km] = {}\n'.format(self.config.sourceDistance))
+            if self._singleMirrorMode:
+                file.write('# mirrorNumber = {}\n\n'.format(self.config.mirrorNumber))
+
+        # Filling in star file with a single star.
+        with self._starsFile.open('w') as file:
+            file.write('0. {} 1.0 {}'.format(
+                90. - self.config.zenithAngle,
```

Comment:
You explained it in the past to me, but just to be sure: you ensure the units somewhere else (e.g., that angles are in degrees)

Reply:
Yes. This is done by our system to validate configurable parameters. Any parameter in self.config was validated and converted to the proper unit.